addons/l10n_it_stock_ddt/models/account_invoice.py

```diff
@@ -0,0 +1,70 @@
+from odoo.tools.float_utils import float_compare
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.move'
+
+    l10n_it_ddt_ids = fields.Many2many('stock.picking', compute="_compute_ddt_ids")
+
+    def _get_ddt_values(self):
+        self.ensure_one()
+        # We don't consider returns/credit notes as they will lead to more deliveries/invoices as well
+        if self.type != "out_invoice":
+            return {}
+        line_count = 0
+        invoice_line_pickings = {}
+        for line in self.invoice_line_ids.filtered(lambda l: not l.display_type and l.move_id.state == 'posted'):
+            line_count += 1
+            done_moves_related = line.sale_line_ids.mapped('move_ids').filtered(lambda m: m.state == 'done' and m.location_dest_id.usage == 'customer')
+            if len(done_moves_related) <= 1:
+                if done_moves_related and line_count not in invoice_line_pickings.get(done_moves_related.picking_id, []):
+                    invoice_line_pickings.setdefault(done_moves_related.picking_id, []).append(line_count)
+            else:
+                total_invoices = done_moves_related.mapped('sale_line_id.invoice_lines').filtered(
+                    lambda m: m.move_id.state == 'posted' and m.move_id.type == 'out_invoice').sorted(lambda m: m.move_id.invoice_date)
+                total_invs = [(i.product_uom_id._compute_quantity(i.quantity, i.product_id.uom_id), i) for i in total_invoices]
+
+                inv = total_invs.pop(0)
+                # Match all moves and related invoice lines FIFO looking for when the matched invoice_line matches line
+                for move in done_moves_related.sorted(lambda m: m.date):
+                    rounding = move.product_uom.rounding
+                    move_qty = move.product_qty
+                    while (float_compare(move_qty, 0, precision_rounding=rounding) > 0):
+                        if float_compare(inv[0], move_qty, precision_rounding=rounding) > 0:
+                            inv = (inv[0] - move_qty, inv[1])
+                            invoice_line = inv[1]
+                            move_qty = 0
+                        if float_compare(inv[0], move_qty, precision_rounding=rounding) <= 0:
+                            move_qty -= inv[0]
+                            invoice_line = inv[1]
+                            if total_invs:
+                                inv = total_invs.pop(0)
+                            else:
+                                move_qty = 0 #abort when not enough matched invoices
+                        # If in our FIFO iteration we stumble upon the line we were checking
+                        if invoice_line == line and line_count not in invoice_line_pickings.get(move.picking_id, []):
+                            invoice_line_pickings.setdefault(move.picking_id, []).append(line_count)
+                    if not total_invs:
+                        break
+        return invoice_line_pickings
```

Comment:
```suggestion
        return invoice_line_pickings.keys()
```

Reply:
indeed I messed up while doing my review, this comment was before I understood the need of both